rust/src/delta_datafusion.rs

```diff
@@ -163,9 +213,51 @@ impl TableProvider for delta::DeltaTable {
     }
 }
 
-fn stat_to_val(stat: &ColumnCountStat) -> Option<usize> {
-    match stat {
-        ColumnCountStat::Value(val) => Some(*val as usize),
-        ColumnCountStat::Column(_) => None,
+fn to_scalar_value(stat_val: &serde_json::Value) -> Option<datafusion::scalar::ScalarValue> {
+    if stat_val.is_number() {
+        if stat_val.is_i64() {
+            Some(ScalarValue::from(stat_val.as_i64()?))
```

Comment:
I think the `is_type()` method call can be avoided if we reuse the option from `as_type` method:

```suggestion
        if let Some(val) = stat_val.as_i64() {
            Some(ScalarValue::from(val))
```